src/hooks/useSchemaValidator.ts

```diff
@@ -0,0 +1,218 @@
+import { useDatatype } from './useDatatype';
+import { useEffect, useMemo, useState } from 'react';
+import { useConfig } from '@openmrs/esm-framework';
+import { ConceptTrue, ConceptFalse } from '../constants';
+import { OHRIFormField, OHRIFormSchema } from '../api/types';
+
+export function useSchemaValidator(schema: OHRIFormSchema) {
```

Comment:
Two things here: 
1. We need a trigger state (to trigger the schema validation)
2. On top of the errors and warnings, it's important to return a progress state i.e.. `isValidating`

Reply:
Noted.